app/renderer/components/navigation/navigator.js

```diff
@@ -169,7 +169,7 @@ class Navigator extends ImmutableComponent {
     if (!e.target.className.includes('navigatorWrapper')) {
       return
     }
-    return !isMaximized() ? currentWindow.maximize() : currentWindow.unmaximize()
+    return !isMaximized() ? windowActions.shouldMaximize(getCurrentWindowId()) : windowActions.shouldMinimize(getCurrentWindowId())
```

Comment:
I guess should* is kind of like *Requested for tab creation, maybe we should use one or the other.  No need to hold this up though just thinking out loud.